synapse/storage/databases/main/events.py

```diff
@@ -2189,8 +2225,10 @@ def _store_rejections_txn(self, txn, event_id, reason):
         )
 
     def _store_event_state_mappings_txn(
-        self, txn, events_and_contexts: Iterable[Tuple[EventBase, EventContext]]
-    ):
+        self,
+        txn: LoggingTransaction,
+        events_and_contexts: Iterable[Tuple[EventBase, EventContext]],
```

Comment:
The current type hint looks wrong since we iterate it multiple times:

```suggestion
        events_and_contexts: Collection[Tuple[EventBase, EventContext]],
```